keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/DeliveryArtifact.kt

```diff
@@ -17,3 +21,37 @@ enum class ArtifactType {
 enum class ArtifactStatus {
   FINAL, CANDIDATE, SNAPSHOT, RELEASE
 }
+
+@JsonTypeInfo(
+  use = Id.NAME,
+  include = As.EXISTING_PROPERTY,
+  property = "type"
+)
+@JsonSubTypes(
+  Type(DebianArtifact::class),
+  Type(DockerArtifact::class)
+)
+interface DeliveryArtifact {
+  val name: String
+  val type: ArtifactType
+  val versioningStrategy: VersioningStrategy
+}
+
+@JsonTypeName("deb")
+data class DebianArtifact(
+  override val name: String,
+  val statuses: List<ArtifactStatus> = emptyList(),
```

Comment:
Side-comment: this got me thinking that these classes don't really represent specific artifacts, per se, but rather a specification for how to _find_ artifacts matching certain attributes. I don't know that it would be worth renaming them, but I find myself confused at times thinking these actually represent specific "snapshot-in-time" versions of an artifact.

Reply:
Oh, yes, you're definitely right about these drifting to represent all the information about how we find an artifact. Given that they're a "delivery artifact" i think it's okay to have that slightly different definition (not a snapshot-in-time artifact). Do you think a class level comment would help explain this?

Reply:
I think it would!